wagtail/core/blocks/field_block.py

```diff
@@ -386,20 +391,22 @@ def __init__(self, choices=None, default=None, required=True, help_text=None, va
             self._constructor_kwargs['required'] = required
         if help_text is not None:
             self._constructor_kwargs['help_text'] = help_text
+        if widget is not None:
+            self._constructor_kwargs['widget'] = widget
 
         # We will need to modify the choices list to insert a blank option, if there isn't
         # one already. We have to do this at render time in the case of callable choices - so rather
         # than having separate code paths for static vs dynamic lists, we'll _always_ pass a callable
         # to ChoiceField to perform this step at render time.
 
         # If we have a default choice and the field is required, we don't need to add a blank option.
```

Comment:
This comment line should be moved into `ChoiceBlock.get_callable_choices`.

Also, I think it's worth us renaming it to `_get_callable_choices` at this point, to clarify that this is an internal method - the fact that it's being inherited / overridden here is liable to get people thinking this is the right place to implement their custom return-a-list-of-choices logic.